src/integrationTest/java/com/mongodb/kafka/connect/avro/SchemaNameIntegrationTest.java

```diff
@@ -0,0 +1,89 @@
+ * limitations under the License.
+ */
+package com.mongodb.kafka.connect.avro;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+
+import io.confluent.connect.avro.AvroData;
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+import io.confluent.kafka.schemaregistry.avro.AvroSchema;
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import com.mongodb.kafka.connect.mongodb.MongoKafkaTestCase;
+
+final class SchemaNameIntegrationTest extends MongoKafkaTestCase {
+  @Test
+  @DisplayName(
+      "Ensure Avro schema namespaces are correctly reflected in Kafka Connect schema names and can be stored in Schema Registry")
+  void schemaName() throws RestClientException, IOException {
+    SchemaRegistryClient src = new CachedSchemaRegistryClient(KAFKA.schemaRegistryUrl(), 1);
+    org.apache.kafka.connect.data.Schema kafkaConnectRecordSchema =
+        com.mongodb.kafka.connect.source.schema.AvroSchema.fromJson(
+            "{\n"
+                + "  \"name\": \"recordSchemaName\",\n"
+                + "  \"namespace\": \"recordSchemaNamespace\",\n"
+                + "  \"type\": \"record\",\n"
+                + "  \"fields\": [\n"
+                + "    {\n"
+                + "      \"name\": \"nestedRecordFieldName\",\n"
+                + "      \"type\": {\n"
+                + "        \"name\": \"nested.Record.Schema.Namespace.nestedRecordSchemaName\",\n"
+                + "        \"namespace\": \"nestedRecordSchemaNamespace\",\n"
+                + "        \"type\": \"record\",\n"
+                + "        \"fields\": [\n"
+                + "          {\n"
+                + "            \"name\": \"fieldName\",\n"
+                + "            \"type\": \"int\"\n"
+                + "          }\n"
+                + "        ]\n"
+                + "      }\n"
+                + "    }\n"
+                + "  ]\n"
+                + "}");
+    org.apache.kafka.connect.data.Schema kafkaConnectNestedRecordFieldSchema =
+        kafkaConnectRecordSchema.field("nestedRecordFieldName").schema();
+    assertAll(
+        () ->
+            assertEquals("recordSchemaNamespace.recordSchemaName", kafkaConnectRecordSchema.name()),
+        () ->
+            assertEquals(
+                "nested.Record.Schema.Namespace.nestedRecordSchemaName",
+                kafkaConnectNestedRecordFieldSchema.name()));
+    org.apache.avro.Schema avroRecordSchema =
+        new AvroData(1).fromConnectSchema(kafkaConnectRecordSchema);
+    int registeredSchemaId = src.register("subject", new AvroSchema(avroRecordSchema));
```

Comment:
This is how `io.confluent.connect.avro.AvroConverter` [converts](https://github.com/confluentinc/schema-registry/blob/a2045b78554b4ae6558f623f818805c52b0d3937/avro-converter/src/main/java/io/confluent/connect/avro/AvroConverter.java#L85) `org.apache.kafka.connect.data.Schema` to `org.apache.avro.Schema` and [stores](https://github.com/confluentinc/schema-registry/blob/a2045b78554b4ae6558f623f818805c52b0d3937/avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerializer.java#L115) it in Schema Registry.